src/components/group.ts

```diff
@@ -52,12 +44,43 @@ export module Components {
     }
 
     public fixedWidth(): boolean {
-      return this.components().every((c) => c.fixedWidth());
+      return this._components().every((c) => c.fixedWidth());
     }
 
     public fixedHeight(): boolean {
-      return this.components().every((c) => c.fixedHeight());
+      return this._components().every((c) => c.fixedHeight());
+    }
+
+    /**
+     * @return {Component[]} The Components in this Group.
+     */
+    public components(): Component[] {
+      return this._componentList.slice();
     }
+
+    protected _components(): Component[] {
+      return this._componentList;
+    }
```

Comment:
Still noting that the above implementation will be really confusing for extenders